### PR TITLE
Fixed Reticle Visibility Bugs and added tests. Cleaning up reticle normal warnings

### PR DESCRIPTION
**Pull request by RogPodge**
## Overview
Fixes a bug where the reticle didn't get disabled when it's visuals script was disabled. Also adds tests around the visibility of FarRayVisuals

Sets the reticle's rotation directly rather than relying on setting transform.forward. Setting transform.forward could result in the following warning being spammed

`Look rotation viewing vector is zero
UnityEngine.Transform:set_forward (UnityEngine.Vector3)

### CODE REVIEW

File: com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKRayReticleVisual.cs [hunk 1]
```diff
@@ -112,18 +112,23 @@ private void UpdateReticle()
...
-                    if (proximityLight != null)
-                    {
-                        proximityLight.SetActive(Reticle.activeSelf);
+                        ReticleSetActive(false);
                     }
                 }
             }
         }
 
+        private void ReticleSetActive(bool value)
```
Comment (MaxWang-MS) [inline com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKRayReticleVisual.cs:121]: Would something like `ReticleAndProxLightSetActive` or `SetActiveReticleAndProxLight` makes more sense?

File: com.microsoft.mrtk.input/Tests/Runtime/FarRayInteractorVisualsTests.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+using Microsoft.MixedReality.Toolkit.Core.Tests;
+using NUnit.Framework;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Input.Tests
+{
+    /// <summary>
+    /// Tests for verifying the behavior of FuzzyGazeInteractor.
```
Comment (MaxWang-MS) [inline com.microsoft.mrtk.input/Tests/Runtime/FarRayInteractorVisualsTests.cs:13]: Is the comment copy paste?

File: com.microsoft.mrtk.input/Tests/Runtime/FarRayInteractorVisualsTests.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    /// </summary>
+    public class FarRayInteractorVisualsTests : BaseRuntimeInputTests
+    {
+        /// <summary>
+        /// Basic test of FuzzyGazeInteractor. Confirm a FuzzyGazeInteractor is active in the scene, and then
```
Comment (MaxWang-MS) [inline com.microsoft.mrtk.input/Tests/Runtime/FarRayInteractorVisualsTests.cs:18]: Same here

File: com.microsoft.mrtk.input/Tests/Runtime/FarRayInteractorVisualsTests.cs (hunk 2 above)
Comment (RogPodge) [reply com.microsoft.mrtk.input/Tests/Runtime/FarRayInteractorVisualsTests.cs:13]: 🍝

File: com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKRayReticleVisual.cs (hunk 1 above)
Comment (RogPodge) [reply com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKRayReticleVisual.cs:121]: prox lights might not always be present so I wanted to focus solely on the reticle aspect with the naming

**Review by MaxWang-MS — APPROVED** [review]

File: com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKRayReticleVisual.cs (hunk 1 above)
Comment (MaxWang-MS) [reply com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKRayReticleVisual.cs:121]: I understand the point, and also the name of a private function does not matter that much. It's just that I was confused when I saw the change from `Reticle.SetActive(true);` to `ReticleSetActive(true);` before reading the content of the function. I am OK either way.

**Review by MaxWang-MS — DISMISSED** [review]

**Review by MaxWang-MS — APPROVED** [review]

**Pull request merged**